### PR TITLE
PBI-3004 Plugin update Android 6.0

### PR DESCRIPTION
**Pull request by ChaseApptentive**
[PBI-3004]

Has Android 6.0.3 in it.
Will update to 6.0.4 when released and there will be very slight differences included at that time.

Added Configuration options for Android
Added Hidden Text message
Updated files to Kotlin
Added utils functionality to ween off legacy dependency

Added requirements for cordova-android level because of this link. They need to do it anyway and it'll help us out if they update to it.
https://ionic.zendesk.com/hc/en-us/articles/7891143965975-Migrating-to-Cordova-Android-11

[PBI-3004]: https://apptentive.atlassian.net/browse/PBI-3004?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,12 +19,10 @@
...
   ],
-  "engines": [
-    {
-      "name": "cordova",
-      "version": ">=3.6.0"
-    }
-  ],
+  "engines": {
+    "cordova": ">=11.0.0",
+    "cordova-android": ">=10.1.1"
```
Comment (ChaseApptentive) [inline package.json:24]: See why this wasn't updated from plugin version

Comment (ChaseApptentive) [reply package.json]: Fixed

File: www/ApptentiveIos.js [hunk 2]
```diff
@@ -98,12 +98,18 @@ var Apptentive = {
...
     login: function (successCallback, errorCallback, token) {
         cordova.exec(successCallback, errorCallback, "ApptentiveBridge", "execute", ["login", token]);
     },
+
     logout: function (successCallback, errorCallback) {
         cordova.exec(successCallback, errorCallback, "ApptentiveBridge", "execute", ["logout"]);
-    }
+    },
+    
+    sendAttachmentText: function (successCallback, errorCallback, text) {
```
Comment (frankus) [inline www/ApptentiveIos.js:110]: Do you know if this is supported in the iOS bridge?

**Review by frankus — APPROVED** [review]
Looks good. Just had to check to see if the iOS bridge code supported the attachment text but it does.

**Review by PoornimaApptentive — APPROVED** [review]

File: src/android/ApptentiveBridge.kt [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+  override fun execute(action: String, args: JSONArray, callbackContext: CallbackContext): Boolean {
+    android.util.Log.d("Apptentive", "[CORDOVA] Executing action: $action")
+
+    val currentActivity = cordova.getActivity()
+    if (currentActivity == null) {
+      android.util.Log.e("Apptentive", "[CORDOVA] Current activity is null. Cannot execute action: $action")
+      callbackContext.error("Current activity is null. Cannot execute action: $action")
+      return true
+    }
```
[the remark this answers is not in the document]
Comment (PoornimaApptentive) [reply src/android/ApptentiveBridge.kt:41]: Just wondering if this have to be for everything. Can we limit to showing the interaction?

[the remark this answers is not in the document]
Comment (PoornimaApptentive) [reply src/android/ApptentiveBridge.kt:41]: I am not sure about cordova activity, in flutter flutter activity won't be available when MC or Survey is active ( I think so)

File: www/ApptentiveIos.js [hunk 4]
```diff
@@ -75,6 +75,10 @@ var Apptentive = {
         cordova.exec(unreadMessagesCallback, errorCallback, "ApptentiveBridge", "execute", ["registerForMessageNotifications"]);
     },
 
+    addSurveyFinishedListener: function (surveyFinishedCallback, errorCallback) {
+        errorCallback(); // Does nothing on iOS.
+    },
```
Comment (ChaseApptentive) [inline www/ApptentiveIos.js:80]: @frankus Do you want to add the functionality for this and sendAttachmentText for iOS for the Cordova app?

Comment (frankus) [reply www/ApptentiveIos.js:80]: I could. I also have a story in the sprint with debug mode.

I'll branch off of your work here and jump on that today.

Comment (frankus) [reply www/ApptentiveIos.js:80]: Added in https://github.com/apptentive/apptentive-cordova/pull/99

**Pull request merged**